nixos/modules/services/backup/borgbackup.nix

```diff
@@ -33,13 +33,26 @@ let
     }
     trap on_exit EXIT
 
+    borgWrapper () {
+      set +e
+      borg "$@"
+      local result=$?
+      set -e
```

Comment:
I think `set` will affect the caller. Is there any downside to running it in a subshell?

Reply:
what do you mean affects the caller? I'm only restoring set -e, so this is expected.
If you meant that set +e line 37 could affect concurrent execution, apparently it does not:
```bash
set -e

fun () {
    set +e
    sleep 2
    false
    echo "false ignored in fun"
    set -e
}

fun2 () {
    sleep 1
    echo "here we know that fun has run set +e"
    false
    echo "we never get here"
}

fun | fun2
```
```
:!bash -x foo.sh
+ set -e
+ fun
+ set +e
+ sleep 2
+ fun2
+ sleep 1
+ echo 'here we know that fun has run set +e'
+ false
here we know that fun has run set +e
+ false
+ echo 'false ignored in fun'

le shell a retourné 1
```

About using a subshell, I think I can't define variables in subshells which are also available in the outer shell.

Reply:
I was thinking about the interaction between the module shell code and the pre/post hook shell code provided by the user, which I think run in the same process. I haven't looked closely, but I'm always cautious about these stateful operations.

Yes, variables defined in sub-shells stay in the sub-shell process, but I was thinking about something like .... hm, you're right, that's difficult to get right with subshells/without parent variables. But how about

```
# either errexit state works with this code, so we don't have to set it at all
#set -e
#set +e

echo "pre-borg"
(exit 1) && rc=$? || rc=$?   # simulate borg exit code
echo "post-borg rc=$rc"
```

Reply:
thanks that works. I changed the PR in favor of this idiom.